### PR TITLE
added the corrosive powder 'barium chloride'

### PR DESCRIPTION
**Pull request by gedrocht**
[Chemistry][Feature]

## About the PR 
This adds the reagent BaCl₂ to the game.

## Why's this needed? 
Currently, there are only 3 reagent recipes that specifically call for Barium, versus 44 that contain Chlorine, 49 that call for Hydrogen, and 52 that call for Oxygen. Mixing Barium (and other chems like Calcium and Helium) with random stuff in-game will currently get you nowhere. I'd like to change that, starting with this. This PR is a proof-of-concept that will inform how such an effort will be received going forward.

## Changelog 

```changelog
(u)gabringis
(+)added barium chloride
```

### CODE REVIEW

**Comment by M-Earthfire** [comment]
Uh, i don't want to be that guy, but just from a chemical side of things: barium chloride is a salt with neutral pH when dissolved in water. It's mostly corrose to some metals due to the containing chloride ions... so as corrosive as table salt... and you coded it as a child of acid, which stuff like sulfuric or nitric acid fall into.

The salt is poisonous as heck because if the containing barium, though. So it could work better as a poison.

File: code/modules/power/TEG.dm [hunk 1]
```diff
@@ -1371,7 +1372,7 @@ Present 	Unscrewed  Connected 	Unconnected		Missing
 /** Thermoelectric Generator Semiconductor - A beautiful array of thermopiles */
 /obj/item/teg_semiconductor
 	name = "Prototype Semiconductor"
-	desc = "A large rectangular plate stamped with 'Prototype Thermo-Electric Generator Semiconductor.  If found please return to NanoTrasen.'"
+	desc = "A large rectangular ae stamped with 'Prototype Thermo-Electric Generator Semiconductor.  If found please return to NanoTrasen.'"
```
Comment (nefarious6th) [inline code/modules/power/TEG.dm:1375]: accidental deletion here? 👁️

**Comment by Exester509** [comment]
> Uh, i don't want to be that guy, but just from a chemical side of things: barium chloride is a salt with neutral pH when dissolved in water. It's mostly corrosive to some metals due to the containing chloride ions... so as corrosive as table salt... and you coded it as a child of acid, which stuff like sulfuric or nitric acid fall into. Well, it is highly irritant due to being hygroscopic in its anhydrous state and toxic nature, but i digress. Msds here:
> 
> https://fscimage.fishersci.com/msds/02370.htm
> 
> The salt is poisonous as heck because if the containing barium, though. So it could work better as a poison.

I could see it being some sort of teargas. Currently we only have capsaicin, and it's almost never used

**Comment by colossusqw** [comment]
> I could see it being some sort of teargas. Currently we only have capsaicin, and it's almost never used

oh, some more dangerous chemical crowd control would be very interesting

**Comment by TDHooligan** [comment]
Was originally a big fan of 'dangerous tear gas' - bigger fan of it with having this new malady, though its' effort (2 chems) to payoff (long-term malady) seems very skewed, as the recipe is super simple. 

perhaps it's scope creep, but could be a fun idea to have the industrial process more closely matched and open the possibility for a second chem to be added later that covers the former suggestion:
Barium + Sulfur + Carbon + Oxygen + Heat = Barium Sulfide (bog standard chem, 0.4 depletion rate, 1 tox or somethin)
Barium Sulfide + Hydrochloric acid = Barium Chloride + (later) Hydrogen Sulfide in the beaker and air

hydrogen sulfide could be a neat mean riot gas, causing losebreath, messages about stink until a critical threshold is inhaled/ingested/elapsed.

**Comment by gedrocht** [comment]
As stated in the initial PR description, this was a proof-of-concept effort to test how the addition of new chemicals would be received. After gauging the response to this single new chemical, I do not believe that several dozen would realistically be accepted. Please feel free to correct me if I'm wrong.

**Comment by M-Earthfire** [comment]
Im not a reviewer, but: 

Firstly, when i look at the PR description, i have no idea what the chemical does. That means i cannot say if i would be interested in adding this chemicals , because i would need to look through the code and see what it does. Noones interested in adding a new chem simply because it is a new chem.

Secondly, until a PR gets reviewed, this neans until someone takes a look at the code, a few weeks can pass. Your PR is just up for two weeks and unable to gather much interest because of constant changes on how it works amd no description. Some PR's of mine took 1,5 months. 

So honestly,  i would chill out, update the description to exactly say what it does and then sit down for 2-3 weeks until somone who can do reviews takes a look at it. If you don't got a reply after a month, try poking a dev.

**Comment by gedrocht** [comment]
You might not agree, but as far as I'm concerned, what's been said so far lines up with what I'm thinking. If it takes weeks to ratify the addition of a single chemical that is this simple, my idea of adding dozens of new chemicals is not feasible for me. This codebase is over 11 million lines. This PR is 88 lines. It has been 15 days. If this was approved today, if this was some impossible, hypothetical, 100% stable rate of approval per lines of code, adding another 11 million lines would take over 5,000 years. If that's how it has to be, that's how it has to be, but I think things would be better if they were some other way, instead.

**Comment by Tarmunora** [comment]
Gonna go with a no on this one - we don't need yet more poisons, and we *really* don't need more generic two-click recipe poisons.

**Comment by gedrocht** [comment]
🤦

**Pull request closed without merging**